dotCMS/src/main/java/com/dotcms/security/apps/ParamDescriptor.java

```diff
@@ -17,17 +20,17 @@ public final class ParamDescriptor extends AbstractProperty<Object>{
 
     private final Boolean required;
 
-    /**
-     * Private Constructor
-     * @param value
-     * @param hidden
-     * @param type
-     * @param label
-     * @param hint
-     * @param required
-     */
-    private ParamDescriptor(final Object value, final Boolean hidden, final Type type, final String label, final String hint, final Boolean required) {
-        super(value, hidden, type);
+    private ParamDescriptor(final Object value,
+                            final Boolean hidden,
+                            final Type type,
+                            final String envVar,
+                            final Boolean envShow,
+                            final String envValue,
+                            final String label,
+                            final String hint,
+                            final Boolean required) {
+        super(value, hidden, type, envVar, envShow);
```

Comment:
Way too many params. Consider a builder here as well. I know this is not new. Maybe at least a TO-DO? but if possible let's refactor

Reply:
Refactored.